inc/analytics.php

```diff
@@ -0,0 +1,205 @@
+	add_action( 'admin_init', $n( 'register_setting' ) );
+	add_filter( 'pantheon.ei.gtm_code', $n( 'filter_gtm_code' ), 1 );
+}
+
+/**
+ * Add the GTM code to the <head> element.
+ */
+function after_head() {
+	$gtm_code = get_gtm_code();
+
+	// Bail early if we don't have a GTM code or if the GTM code is being overridden externally.
+	if ( is_bool( $gtm_code ) ) {
+		return;
+	}
+	?>
+<!-- Google Tag Manager -->
+<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+})(window,document,'script','dataLayer','<?php echo esc_attr( $gtm_code ); ?>');</script>
+<!-- End Google Tag Manager -->
+	<?php
+}
+
+/**
+ * Add the GTM code to the <body> element.
+ */
+function after_body() {
+	$gtm_code = get_gtm_code();
+
+	// Bail early if we don't have a GTM code or if the GTM code is being overridden externally.
+	if ( is_bool( $gtm_code ) ) {
+		return;
+	}
+	?>
+<!-- Google Tag Manager (noscript) -->
+<noscript><iframe src="https://www.googletagmanager.com/ns.html?id=<?php echo esc_attr( $gtm_code ); ?>"
+height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
+<!-- End Google Tag Manager (noscript) -->
+	<?php
+}
+
+/**
+ * Localize the GTM script.
+ */
+function localize_script() {
+	$vary_headers = WP\get_supported_vary_headers();
+
+	wp_localize_script( 'pantheon-ei', 'eiGtm', [
+		'headersEnabled' => $vary_headers,
+		'geo' => WP\Geo\get_geo(),
+		'interest' => WP\Interest\get_interest(),
+	] );
+}
+
+/**
+ * Get the GTM code.
+ * If no GTM code exists, return false.
+ * Alternately, the user can filter this value as "true" if they want to use their own Google Analytics/Google Tag Manager plugin to store/manage their GTM code.
+ *
+ * @return bool|string The GTM code, or false if none exists.
+ */
+function get_gtm_code() {
+	$gtm_code = apply_filters( 'pantheon.ei.gtm_code', false );
+
+	// Return false if we don't have a valid GTM code.
+	if ( ! $gtm_code || is_string( $gtm_code ) && false === stripos( $gtm_code, 'GTM' ) ) {
+		return false;
+	}
+
+	return $gtm_code;
+}
+
+/**
+ * Register the GTM setting and add the setting field.
+ */
+function register_setting() {
+	$gtm_code = get_gtm_code();
+
+	// Bail early and don't register the setting if we're overriding GTM in the EI plugin.
+	if ( true === $gtm_code ) {
+		return;
+	}
+
+	// Get the option value so we can check if GTM is set in the plugin or via a filter.
+	$option = get_option( 'pantheon_ei_gtm_code' );
+
+	// If the $gtm_code is a string, but we don't have an option, it was added via filter, so bail here and don't display the option in the admin.
+	if ( is_string( $gtm_code ) && ! $option ) {
+		return;
+	}
+
+	// Regiter the setting.
+	\register_setting( 'general', 'pantheon_ei_gtm_code', [
```

Comment:
```suggestion
	register_setting( 'general', 'pantheon_ei_gtm_code', [
```

Reply:
A possible solution would be renaming the function to something like `register_gtm_setting` or `resister_settings` if it's an issue. Because other settings might be added later, though, I didn't want to be more specific about the setting name being registered.

Reply:
renamed function to `register_ei_settings`